### ISSUE TITLE
`SeparateDefinitionBlocks` inserts a new line before `// clang-format on` when it's directly beneath a concept

### ISSUE THREAD

**Issue by cjdb** [llvm/llvm-project#63393]
`clang-format test.cpp --style="{Language: Cpp, Standard: Latest, SeparateDefinitionBlocks: Always, }"` transforms this
```cpp
// clang-format off
template<class T>
concept C = not A<S<T>>;
// clang-format on

template<class>
struct E {};
```
into this
```cpp
// clang-format off
template<class T>
concept C = not A<S<T>>;

// clang-format on

template <class> struct E {};
```
Built using 0e7cfa59ebe9.

**Comment by rymiel** [comment]
I agree it's a bug, but invalid-code-generation applies for "Tool produced invalid code that no longer compiles", which doesn't seem to be the case here?

**Comment by cjdb** [comment]
Hmm, I've seen it used for still-compiling-but-malformed clang-format output before. You're right though.

**Comment by owenca** [comment]
https://reviews.llvm.org/D156971

**Comment by owenca** [comment]
Fixed in 58a71c66db85f65bb6cdefe314f62318b95e9fc4.
